pkgs/applications/blockchains/lighthouse/default.nix

```diff
@@ -0,0 +1,86 @@
+{ lib
+, perl
+, fetchurl
+, stdenv
+, clang
+, cmake
+, darwin
+, protobuf
+, rustPlatform
+, llvmPackages
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "2.5.1";
+
+  src = fetchFromGitHub {
+    owner = "sigp";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-o8fntnEDRRtxDHsqHTJ3GoHpFtRpuwPQQJ+kxHV1NKA=";
+  };
+
+  cargoHash = "sha256-oyX+v2dJ54GXd0XdesZC/zw7FXtFfx6qOJgX4OJLR+A=";
+
+  # Do not run tests, because currently some tests require ganache (https://www.npmjs.com/package/ganache),
+  # and I haven't figured out how to import ganache into Nixpkgs.
+  doCheck = false;
+
+  cargoBuildFlags = [
+    # Make only the `lighthouse` executable. Without this, the package produces 7 executables.
+    "-p lighthouse"
+  ];
+
+  nativeBuildInputs = [
+    # https://lighthouse-book.sigmaprime.io/installation-source.html
+    clang
+    cmake
+    llvmPackages.libclang
```

Comment:
I am not certain `llvmPackages.libclang` is required here, since it is referenced below.